sqlmesh/core/plan/builder.py

```diff
@@ -107,6 +107,18 @@ def __init__(
         ensure_finalized_snapshots: bool = False,
         interval_end_per_model: t.Optional[t.Dict[str, int]] = None,
         console: t.Optional[PlanBuilderConsole] = None,
+        user_defined_flags: t.Optional[
+            t.Dict[
+                str,
+                t.Union[
+                    t.Optional[TimeLike],
```

Comment:
Please let's use a type alias + see my other comment wrt optional and iterable vs collection